src/ops/checkout.rs

```diff
@@ -64,3 +67,60 @@ fn checkout_new_branch_prompt_update(state: &mut State, term: &mut Term) -> Res<
     }
     Ok(())
 }
+
+pub(crate) struct DeleteBranch;
+impl OpTrait for DeleteBranch {
+    fn get_action(&self, target: Option<&TargetData>) -> Option<Action> {
+        match target {
+            Some(TargetData::Branch(branch)) => {
+                let branch = branch.clone();
+                Some(create_y_n_prompt(
+                    Rc::new(move |state, term| delete_branch(state, term, &branch)),
+                    "Delete this branch?",
+                ))
+            }
+            _ => Some(create_prompt_with_default(
+                "Delete branch",
+                delete_branch,
+                selected_rev,
+                true,
+            )),
+        }
+    }
+
+    fn display(&self, _state: &State) -> String {
+        "Delete branch".into()
+    }
+}
+
+fn delete_branch(state: &mut State, term: &mut Term, branch: &str) -> Res<()> {
+    if branch.is_empty() {
+        return Err(Error::InvalidBranch);
+    }
+
+    let delete_flag = {
+        let current_branch = crate::git::get_current_branch(&state.repo)?;
+        let current_branch_name = match current_branch.name() {
+            Ok(Some(name)) => name,
+            Ok(None) => return Err(Error::CantGetBranchName),
+            Err(e) => return Err(Error::CantGetBranch(e)),
+        };
+
+        if branch == current_branch_name {
+            return Err(Error::CantDeleteCurrentBranch);
+        }
+
+        let target_branch = crate::git::get_branch(&state.repo, branch)?;
+        if crate::git::is_branch_unmerged(&state.repo, &target_branch).unwrap_or(false) {
+            "-D"
+        } else {
+            "-d"
+        }
+    };
+
+    let mut cmd = Command::new("git");
+    cmd.args(["branch", delete_flag, branch]);
```

Comment:
```suggestion
    cmd.args(["branch", "-D", branch]);
```

I think this'll work too!